contracts/contracts/vault/OETHVaultCore.sol

```diff
@@ -1,26 +1,147 @@
 // SPDX-License-Identifier: MIT
 pragma solidity ^0.8.0;
 
+import { StableMath } from "../utils/StableMath.sol";
 import { VaultCore } from "./VaultCore.sol";
 
+import { SafeERC20 } from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import { IStrategy } from "../interfaces/IStrategy.sol";
+
 /**
  * @title OETH VaultCore Contract
  * @author Origin Protocol Inc
  */
 contract OETHVaultCore is VaultCore {
+    using SafeERC20 for IERC20;
+    using StableMath for uint256;
+
     address public immutable weth;
+    uint256 public wethAssetIndex;
 
     constructor(address _weth) {
         weth = _weth;
     }
 
+    /**
+     * @dev Caches WETH's index in `allAssets` variable.
+     *      Reduces gas usage by redeem by caching that.
+     */
+    function cacheWETHAssetIndex() external onlyGovernor {
+        uint256 assetCount = allAssets.length;
+        for (uint256 i = 0; i < assetCount; ++i) {
+            if (allAssets[i] == weth) {
+                wethAssetIndex = i;
+                break;
+            }
+        }
+
+        require(allAssets[wethAssetIndex] == weth, "Invalid WETH Asset Index");
+    }
+
     // @inheritdoc VaultCore
-    function mint(
+    function _mint(
         address _asset,
         uint256 _amount,
         uint256 _minimumOusdAmount
-    ) external virtual override whenNotCapitalPaused nonReentrant {
+    ) internal virtual override {
         require(_asset == weth, "Unsupported asset for minting");
```

Comment:
We've removed the check that this is an asset marked as supported. This check is indeed not needed for both minting and redeeming with hardcoded WETH. asset.isSupported is also ignored in balance checks as well.

I think we should just add a note to the asset.isSupported struct parameter that it is ignored in minting, redeeming, and balance checking on OETH.

Reply:
Have added the comment in `VaultStorage` where the `Asset` struct is defined